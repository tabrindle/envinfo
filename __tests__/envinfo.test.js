const envinfo = require('../src/envinfo');
const helpers = require('../src/helpers');

jest.mock('../src/helpers');

// cycle through the helperFns and mock them according to their name
Object.keys(helpers).forEach(helperFn => {
  const match = helperFn.match(/get(.*)Info/);
  if (match) {
    const name = match[1];
    helpers[helperFn].mockImplementation(() =>
      Promise.resolve([name, '10.0.0', `/usr/local/bin/${name.toLowerCase()}`])
    );
  }
});

describe('Running the programmatic interface', () => {
  test('returns expected json value', () => {
    return envinfo.run({ Binaries: ['Node'] }, { json: true }).then(data => {
      expect(JSON.parse(data)).toMatchSnapshot();
    });
  });

  test('returns expected yaml value', () => {
    return envinfo.run({ Binaries: ['Node'] }).then(data => {
      expect(data).toMatchSnapshot();
    });
  });

  test('returns expected markdown value', () => {
    return envinfo.run({ Binaries: ['npm'] }, { markdown: true }).then(data => {
      expect(data).toMatchSnapshot();
    });
  });

  test('returns expected json value with multiple values', () => {
    return envinfo
      .run(
        {
          Binaries: ['Node', 'Yarn', 'npm'],
        },
        { json: true }
      )
      .then(data => {
        expect(JSON.parse(data)).toMatchSnapshot();
      });
  });

  test('returns expected json value with multiple categories', () => {
    return envinfo
      .run(
        {
          Binaries: ['Node'],
          Languages: ['Bash'],
        },
        { json: true }
      )
      .then(data => {
        expect(JSON.parse(data)).toMatchSnapshot();
      });
  });

  test('filters out returned values with N/A', () => {
    helpers.getChromeInfo.mockImplementation(() => Promise.resolve(['Chrome', 'N/A', 'N/A']));

    return envinfo.run({ Browsers: ['Chrome', 'Firefox'] }, { json: true }).then(data => {
      expect(JSON.parse(data)).toMatchSnapshot();
    });
  });

  test('filters out returned path values with N/A', () => {
    helpers.getChromeInfo.mockImplementation(() =>
      Promise.resolve(['Chrome', '65.0.3325.181', 'N/A'])
    );

    return envinfo.run({ Browsers: ['Chrome'] }, { json: true }).then(data => {
      return expect(JSON.parse(data)).toEqual({
        Browsers: {
          Chrome: { version: '65.0.3325.181' },
        },
      });
    });
  });

  test('returns expected title in json', () => {
    return envinfo
      .run({ Binaries: ['Node'] }, { title: 'envinfo rocks!', json: true })
      .then(data => {
        expect(JSON.parse(data)).toMatchSnapshot();
      });
  });

  test('returns expected title in yaml', () => {
    return envinfo.run({ Binaries: ['Node'] }, { title: 'envinfo rocks!' }).then(data => {
      expect(data).toMatchSnapshot();
    });
  });
});
