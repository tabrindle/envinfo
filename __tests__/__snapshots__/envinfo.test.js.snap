// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Running the programmatic interface filters out returned values with N/A 1`] = `
Object {
  "Browsers": Object {
    "Firefox": Object {
      "path": "/usr/local/bin/firefox",
      "version": "10.0.0",
    },
  },
}
`;

exports[`Running the programmatic interface returns expected json value 1`] = `
Object {
  "Binaries": Object {
    "Node": Object {
      "path": "/usr/local/bin/node",
      "version": "10.0.0",
    },
  },
}
`;

exports[`Running the programmatic interface returns expected json value with multiple categories 1`] = `
Object {
  "Binaries": Object {
    "Node": Object {
      "path": "/usr/local/bin/node",
      "version": "10.0.0",
    },
  },
  "Languages": Object {
    "Bash": Object {
      "path": "/usr/local/bin/bash",
      "version": "10.0.0",
    },
  },
}
`;

exports[`Running the programmatic interface returns expected json value with multiple values 1`] = `
Object {
  "Binaries": Object {
    "Node": Object {
      "path": "/usr/local/bin/node",
      "version": "10.0.0",
    },
    "Yarn": Object {
      "path": "/usr/local/bin/yarn",
      "version": "10.0.0",
    },
    "npm": Object {
      "path": "/usr/local/bin/npm",
      "version": "10.0.0",
    },
  },
}
`;

exports[`Running the programmatic interface returns expected markdown value 1`] = `
"
## Binaries:
 - npm: 10.0.0 - /usr/local/bin/npm
"
`;

exports[`Running the programmatic interface returns expected title in json 1`] = `
Object {
  "envinfo rocks!": Object {
    "Binaries": Object {
      "Node": Object {
        "path": "/usr/local/bin/node",
        "version": "10.0.0",
      },
    },
  },
}
`;

exports[`Running the programmatic interface returns expected title in yaml 1`] = `
"
  envinfo rocks!:
    Binaries:
      Node: 10.0.0 - /usr/local/bin/node
"
`;

exports[`Running the programmatic interface returns expected yaml value 1`] = `
"
  Binaries:
    Node: 10.0.0 - /usr/local/bin/node
"
`;
